test/GPv2Encoding.test.ts

```diff
@@ -233,15 +235,15 @@ describe("GPv2Encoding", () => {
       const encoder = new SettlementEncoder(testDomain);
       await encoder.signEncodeTrade(
         sampleOrder,
-        sampleOrder.sellAmount,
+        {},
         traders[0],
         SigningScheme.TYPED_DATA,
       );
 
       // NOTE: `v` must be either `27` or `28`, so just set it to something else
       // to generate an invalid signature.
       const encodedTradeBytes = ethers.utils.arrayify(encoder.encodedTrades);
-      encodedTradeBytes[139] = 42;
+      encodedTradeBytes[141] = 42;
```

Comment:
Wow, how did you remember to do this? I'd expect that the test fails even without this change.

Reply:
Yes, the unit test caught this :tada:.

>  I'd expect that the test fails even without this change.

No, it just recovers to wrong address. In general, if you have a signed message and change the message value, the `ecrecover` still works (in fact I have not been able to make it not work), it just recovers to a random address.